platform/mock-local-home-platform.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * Mock Local Home Platform class
+ * - Interacts with bundled HomeApp
+ * - Accepts scan config
+ * - Interacts with virtual network
+ */
+
+/// <reference types="@google/local-home-sdk" />
+import { MockNetwork, MockUDPListener, RemoteAddressInfo } from './mock-radio';
+
+export enum ScanState {
+  Unprovisioned,
+  Provisioned,
+}
+
+interface ScanConfig {
+  state: ScanState;
+}
+
+export class UDPScanConfig implements ScanConfig {
+  state: ScanState;
+  broadcastAddress: string;
+  listenPort: number;
+  broadcastPort: number;
+  discoveryPacket: string;
+  constructor(
+    state: ScanState,
+    broadcastAddress: string,
+    listenport: number,
+    broadcastPort: number,
+    discoveryPacket: string
+  ) {
+    this.state = state;
+    this.broadcastAddress = broadcastAddress;
+    this.broadcastPort = broadcastPort;
+    this.listenPort = listenport;
+    this.discoveryPacket = discoveryPacket;
+  }
+}
+
+// TODO(cjdaly): figure out how to direct arbitrary instances of stub classes to an arbitrary
+// instance of MockLocalHomePlatform. May need to use some static event or singleton pattern.
+
+// TODO(cjdaly): add other radio scan support
+export class MockLocalHomePlatform implements MockUDPListener {
+  private udpScanConfigs: UDPScanConfig[] = [];
+  private network: MockNetwork;
+  private localDeviceIds: string[] = [];
+
+  constructor(network: MockNetwork, udpScanConfigs: UDPScanConfig[]) {
+    this.network = network;
+    this.udpScanConfigs = udpScanConfigs;
+    this.setupUDP();
+  }
+
+  public getLocalDeviceIds(): string[] {
+    return this.localDeviceIds;
+  }
+
+  private setupUDP() {
+    this.udpScanConfigs.forEach((udpScanConfig) => {
+      this.network.registerUDPListener(
+        this,
+        udpScanConfig.listenPort,
+        udpScanConfig.broadcastAddress
+      );
+    });
+  }
+
+  private sendUDPBroadcast(scanConfig: UDPScanConfig) {
+    const packetBuffer = Buffer.from(scanConfig.discoveryPacket, 'hex');
+    this.network.sendUDPMessage(
+      packetBuffer,
+      scanConfig.broadcastPort,
+      scanConfig.broadcastAddress,
+      scanConfig.listenPort,
+      scanConfig.broadcastAddress
+    );
+  }
+
+  // Establish fulfillment path using app code
+  onUDPMessage(msg: Buffer, rinfo: RemoteAddressInfo): void {
+    console.log('received discovery payload:', msg, 'from:', rinfo);
+    const identifyRequest: string = JSON.stringify({
```

Comment:
shouldn't that call the fulfliment somehow?

Reply:
Yes, right now it just assumes it received a discovery response.  In a future PR I plan to implement fulfillment in this method.